documentation/configuration.md

```diff
@@ -0,0 +1,266 @@
+# Configuration
+
+`dotnet-monitor` has extensive configuration to control various aspects of it's behavior. Ordinarily, you are not required to specify most of this configuration and only exists if you wish the change the default behavior in `dotnet-monitor`.
+
+## Configuration Sources
+
+`dotnet-monitor` can read and combine configuration from multiple sources. The configuration sources are listed below in the order in which they are read (Environment variables are highest precedence) :
+
+- Command line parameters
+- User settings path
+  - On Windows, `%USERPROFILE%\.dotnet-monitor\settings.json`
+  - On \*nix, `$XDG_CONFIG_HOME/dotnet-monitor/settings.json`
+  -  If `$XDG_CONFIG_HOME` isn't defined, we fall back to ` $HOME/.config/dotnet-monitor/settings.json`
+- [Key-per-file](https://docs.microsoft.com/aspnet/core/fundamentals/configuration/#key-per-file-configuration-provider) in the shared settings path
+    - On Windows, `%ProgramData%\dotnet-monitor`
+    - On \*nix, `etc/dotnet-monitor`
+
+- Environment variables
+
+### Translating configuration between providers
+
+While the rest of this document will showcase configuration examples in a json format, the same configuration can be expressed via any of the other configuration sources. For example, the API Key configuration can be expressed via shown below:
+
+```json
+{
+  "ApiAuthentication": {
+    "ApiKeyHash": "CB233C3BE9F650146CFCA81D7AA608E3A3865D7313016DFA02DAF82A2505C683",
+    "ApiKeyHashType": "SHA256"
+  }
+}
+```
+
+The same configuration can be expressed via environment variables using the `DotnetMonitor_` prefix and using `__`(double underscore) as the hierarchical separator
+
+```bash
+export DotnetMonitor_ApiAuthentication__ApiKeyHash="CB233C3BE9F650146CFCA81D7AA608E3A3865D7313016DFA02DAF82A2505C683"
+export DotnetMonitor_ApiAuthentication__ApiKeyHashType="SHA256"
+```
+
+#### Kubernetes
+
+When running in Kubernetes, you are able to specify the same configuration via Kubernetes secrets (or configuration maps)
+
+```bash
+kubectl create secret generic apikey \
+  --from-literal=ApiAuthentication__ApiKeyHash=$hash \
+  --from-literal=ApiAuthentication__ApiKeyHashType=SHA256 \
+  --dry-run=client -o yaml \
+  | kubectl apply -f -
+```
+
+You can then use a Kubernetes volume mount to supply the secret to the container at runtime
+
+```yaml 
+spec:
+  volumes:
+  - name: apikey
+    secret:
+      secretName: apikey
+  containers:
+  - name: dotnetmonitoragent
```

Comment:
Did you want to do a projected volume example here, to merge config-maps and secrets?

Reply:
This could also act as a recommended guidance around which configuration elements to store as secrets rather than simple files or config maps.

Reply:
Yup. I'll add in an example merging config maps and secrets.